src/utils/validation/retryUntilValid.js

```diff
@@ -0,0 +1,14 @@
+import output from "../../view/output.js";
+
+export const retryUntilValid = async (getInputFn, parseFn, validator) => {
+  while (true) {
+    try {
+      const userInput = await getInputFn();
+      const result = parseFn(userInput);
+      validator(result);
+      return result;
+    } catch (error) {
+      output.print(error.message);
+    }
+  }
+};
```

Comment:
이 함수를 조금 더 범용적으로 만들 수 있을 것 같아요. 
목표는 while(true) + try/catch에 포커스를 맞춰서 분리하는건데요~!

try문 내부의 로직이 통째로 1개의 함수라고 생각해봅시다. 

```suggestion
export const retryUntilValid = async (func) => {
  while (true) {
    try {
      return func()
    } catch (error) {
      output.print(error.message);
    }
  }
};
```
이렇게 개선하면 사용처에서는 원래 이 함수에서 하던 동작을 함수 하나로 묶어 넣어주면 되겠죠. (input 받고, validation 하고, validation이 끝난 input을 return하는 함수)

Reply:
위 함수도 여전히 개선점이 있는데요~!
`output` view 로직과 결합되어있다는 점이에요.

``` js
export const retryUntilValid = async (func, onError) => {
  while (true) {
    try {
      return func()
    } catch (error) {
      onError(error);
    }
  }
};
```

이런식으로 `에러시에 error객체를 인자로 받아 실행 할 함수` 를 인자로 받도록 하면 output 과의 결합도 끊어질 것 같네요.

Reply:
함수의 인자를 2개로 제한했던 프로그래밍 규칙은 어떤 의도로 존재했을지 한번 더 생각해보시고, 
위 코멘트를 차근히 순서대로 적용해보며 이해해봅시다😄
(꼭 생각해보셔요!)

Reply:
범용적으로 사용하기 위해 이렇게 utils의 공용함수들을 구체적이지 않고 인자로 입력받아서 활용할 수 있도록 하는 것이 활용처가 더 늘어날 수 있겠다는 생각이 들었습니다. utils와 view로직이 결합되어 있게 되면 step2에서 웹으로 제작할때도, 터미널로 출력하는 로직이 있기때문에 활용하기 어려울 수 있겠다 싶어서 [수정](https://github.com/woowacourse/javascript-lotto/pull/358/commits/3fae941d0bc325b8795bb2a25fe22c14297ce3f6)해보았습니다!